src/endpoints/achievements.ts

```diff
@@ -0,0 +1,164 @@
+import { AbstractEndpoint } from '../endpoint'
+import { Schema } from './schemas/schema'
+
+
+type Flag = 'PvE' | 'PvP' | 'WvW' | 'SpecialEvent'
+type Product = 'GuildWars2' | 'HeartOfThorns' | 'PathOfFire' | 'EndOfDragons'
+type Condition = 'HasAccess' | 'NoAccess'
+type LevelTuple = [number, number]
+type LevelObj = { min: number, max: number }
+type RequiredAccess = { product: Product, condition: Condition }
+
+export namespace SchemaOld {
+  /** {@link https://wiki.guildwars2.com/wiki/API:2/achievements/daily} */
+  type Daily = {
+    id: number,
+    level: LevelObj,
+    required_access: Product[]
```

Comment:
Another small nit, but that's something that can be done at the very end as well: I use `Array<Product>` over `Product[]` (because I think it's a bit more readable without having to parse back)

Reply:
sure! I'd like to point out the increasing nesting of structures like

```ts
Promise<Array<number>>
```

but I have no strong bias either way. Will change.

Reply:
Regarding this change and the one about converting `require` to `import`: there are ESLint rules for both those changes available. I noticed you have not employed ESLint so far. Would you be opposed to adding it in?

Reply:
Yeah the nesting (IMO) still reads more natural tho ("a promise of an array of numbers")

ESLint: Go for it! I have a baseline here https://github.com/devoxa/eslint-config but feel free to just configure those two as a start.